src/Farmer/Arm/Dns.fs

```diff
@@ -0,0 +1,155 @@
+let soaRecord = ResourceType ("Microsoft.Network/dnszones/SOA", "2018-05-01")
+let srvRecord = ResourceType ("Microsoft.Network/dnszones/SRV", "2018-05-01")
+let ptrRecord = ResourceType ("Microsoft.Network/dnszones/PTR", "2018-05-01")
+
+type DnsRecordType with
+    member this.ResourceType =
+        match this with
+        | Unknown -> failwith "Not Implemented"
+        | CName -> cnameRecord
+        | A -> aRecord
+        | AAAA -> aaaaRecord
+        | NS -> nsRecord
+        | PTR -> ptrRecord
+        | TXT -> txtRecord
+        | MX -> mxRecord
+        // | Soa -> soaRecord
+        // | Srv -> srvRecord
+
+type DnsZone =
+    { Name : ResourceName
+      Properties : {| ZoneType : string |} }
+
+    interface IArmResource with
+        member this.ResourceName = this.Name
+        member this.JsonModel =
+            {| zones.Create(this.Name, Location.Global) with
+                properties = {| zoneType = this.Properties.ZoneType |}
+            |} :> _
+
+module DnsRecords =
+    type CNameDnsRecord =
+        { Name : ResourceName
+          Zone : ResourceName
+          Type : DnsRecordType
+          TTL : int
+          TargetResource : ResourceName option
+          CNameRecord : string option }
+        interface IArmResource with
+            member this.ResourceName = this.Name
+            member this.JsonModel =
+                {| this.Type.ResourceType.Create(this.Zone + this.Name, dependsOn = [ this.Zone ]) with
+                    properties =
+                        {| TTL = this.TTL
+                           targetResource = {| id = this.TargetResource |}
+                           CNAMERecords = {| cname = this.CNameRecord |> Option.toObj |} |}
+                |} :> _
+
+    type MxDnsRecord =
+        { Name : ResourceName
+          Zone : ResourceName
+          Type : DnsRecordType
+          TTL : int
+          MxRecords : {| Preference : int; Exchange : string |} list }
+        interface IArmResource with
+            member this.ResourceName = this.Name
+            member this.JsonModel =
+                {| this.Type.ResourceType.Create(this.Zone + this.Name, dependsOn = [ this.Zone ]) with
+                    properties =
+                        {| TTL = this.TTL;
+                           MXRecords = this.MxRecords |> List.map (fun mx -> {| preference = mx.Preference; exchange = mx.Exchange |}) |}
+                |} :> _
+
+    type NsDnsRecord =
+        { Name : ResourceName
+          Zone : ResourceName
+          Type : DnsRecordType
+          TTL : int
+          NsRecords : string list }
+        interface IArmResource with
+            member this.ResourceName = this.Name
+            member this.JsonModel =
+                {| this.Type.ResourceType.Create(this.Zone + this.Name, dependsOn = [ this.Zone ]) with
+                    properties =
+                        {| TTL = this.TTL
+                           targetResource = {| id = null |}
```

Comment:
This field won't be emitted - JSON .NET is set up to remove any null values.

Reply:
Removed